### PR TITLE
Move coprocessor test utilites to test_coprocessor library

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

This is a *prelude* PR for the upcoming PR to fix a coprocessor issue, which needs to call coprocessor requests in failpoint tests. So, this PR extracts these common coprocessor test utilities into a workspace called test_coprocessor, just like other test utilities.

Additionally, it also made some changes to the coprocessor test utilities:
- Accepts a read pool configuration
- Accepts an endpoint scheduler instead of endpoint worker (so that it can be used in threads)

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

CI

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

**Comment by breeswish** [comment]
This PR is very simple, it only moves test code and made very simple changes. PTAL @hicqu @zhangjinpeng1987

File: tests/coprocessor/test_select.rs [hunk 1]
```diff
@@ -838,7 +126,15 @@ fn test_stream_batch_row_limit() {
...
         cfg.end_point_stream_batch_row_limit = stream_row_limit;
-        init_data_with_details(Context::new(), engine, &product, &data, true, cfg)
+        init_data_with_details(
+            Context::new(),
+            engine,
+            &product,
+            &data,
+            true,
+            &cfg,
+            &readpool::Config::default_for_test(),
```
Comment (overvenus) [inline tests/coprocessor/test_select.rs]: Why change to `readpool::Config::default_for_test`?

File: components/test_coprocessor/dag.rs [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+use tipb::expression::{ByItem, Expr, ExprType};
+use tipb::schema::ColumnInfo;
+use tipb::select::{Chunk, DAGRequest};
+
+use tikv::coprocessor::codec::{datum, table, Datum};
+use tikv::coprocessor::REQ_TYPE_DAG;
+use tikv::util::codec::number::NumberEncoder;
+
+pub struct DAGSelect {
+    pub execs: Vec<Executor>,
```
Comment (overvenus) [inline components/test_coprocessor/dag.rs]: Could you private these fileds?

Comment (breeswish) [reply components/test_coprocessor/dag.rs]: they are accessed in test_select directly

File: tests/coprocessor/test_select.rs (hunk 1 above)
Comment (breeswish) [reply tests/coprocessor/test_select.rs]: Previously it is default_for_test as well, this PR just makes it configurable in the underlying function from arguments.

**Comment by breeswish** [comment]
@siddontang Build fixed. Please approve again, thanks!

/run-integration-tests

**Comment by siddontang** [comment]
PTAL @overvenus

**Pull request merged**